versioning/src/main/scala/org/typelevel/sbt/TypelevelVersioningPlugin.scala

```diff
@@ -37,6 +38,12 @@ object TypelevelVersioningPlugin extends AutoPlugin {
     lazy val tlUntaggedAreSnapshots =
       settingKey[Boolean](
         "If true, an untagged commit is given a snapshot version, e.g. 0.4-00218f9-SNAPSHOT. If false, it is given a release version, e.g. 0.4-00218f9. (default: true)")
+
+    lazy val tlLatestVersion = settingKey[Option[String]](
+      "The latest tagged version on this branch. Priority is given to the latest stable version, but if you have tagged a binary-breaking prelease version (such as a milestone or release candidate), that will be selected instead. If applicable, this will be the current tagged version.")
+
+    lazy val tlLatestPreReleaseVersion = settingKey[Option[String]](
+      "The latest tagged version on this branch, including milestones and release candidates. If applicable, this will be the current tagged version.")
```

Comment:
ok, @bplommer, what do you think? phew 😓

Reply:
I'm happy with that 👍

Reply:
On second thought, what do you think about `tlLatestVersionOnBranch` and `tlLatestPreReleaseVersionOnBranch`? It's a bit wordy, but for me it's really counterintuitive that (for example) these would be 0.22 versions when running on the http4s 0.22.x branch, so they're not really the latest versions.

Reply:
> but for me it's really counterintuitive that (for example) these would be 0.22 versions when running on the http4s 0.22.x branch

Ok yeah that's super fair. But that is quite a mouthful. What if we used the terminology "previous version" instead of "latest"? Then maybe it would be more clear it's relative to our current context. The problem is that it muddies the fact that this can be the _current_ version it the current commit is tagged.

Reply:
I don't like "previous", for the reason you give - to me that explicitly rules out it being the current one. But I don't have any other suggestions at the moment 🤔

Reply:
Blah. I think I'm just going to run with the current naming. I think reasonable for the majority of cases (esp. for unbranched projects), and the docs cover it.

> but for me it's really counterintuitive that (for example) these would be 0.22 versions when running on the http4s 0.22.x branch

I guess my question is, what are you doing with `tlLatestVersion` on the 0.22.x branch, where you want it to be an 0.23+ version? My assumption is if you've cut off a series branch, you want to "freeze" it in a sense. So although the setting name doesn't reflect it, I think the actual behavior is intuitive.

Reply:
> I guess my question is, what are you doing with tlLatestVersion on the 0.22.x branch, where you want it to be an 0.23+ version?

I can't think of anything! But I *can* imagine wanting the "latest prerelease" version on the 0.23.x branch to be a 1.0 milestone, so I could populate the website's landing page accordingly.

Reply:
In that case, shouldn't the landing page be published from `main` instead of `series/0.23`? (This is how we do it in http4s).

Reply:
Hmm, I guess not necessarily. My proposal in https://github.com/typelevel/sbt-typelevel/pull/317 aims to reduce that complexity, so that you only publish the website from the stable branch. Blah again 😂

Reply:
Blah indeed. Let's just go with it as it is, I think it's good enoigh?

Reply:
Yolo. We might end up adding a couple more of these as the complaints roll in 😂 thank you so much for your work on this.